File: isprime/isprime.py
#!/bin/python
import math

def isprime(n):
	if n<2:
		return False
	elif n==2:
		return True
	else:
		h = n/2+1
		for i in range(2, h):
			r = n%i
			if r==0:
				return False
		return True

def isprime_ex1(x):
	if x<2: return False
	if x==2:return True

	s = math.sqrt(x)
	m = int(s+1)
	t = range(2, m)
	#print t
	for i in t:
		if x%i==0:
			return False

	return True


n = input("input an integer:")
a = range(2, n)
o = filter(isprime_ex1, a)

print o

#for i in range(1,n):
#	b = isprime_ex1(i)
#	if b:
#		print(i),
#		print(' is prime')
#	else:
#		print(' is composite')

File: isprime/isprime.c
#include <stdio.h>
#include <stdbool.h>

bool isprime(long n);

int main(int argc, char**argv)
{
	long n = 0;
	long i;
	bool b;
	printf("input an integer:");
	scanf("%lld",&n);
	printf("\n");

	for(i=2; i<n; i++)
	{
		if(isprime(i))
		{
			printf("%lld is prime.\n", i);
		}
		else
		{
			printf("%lld is composite.\n", i);
		}
	}
	return 0;
}

bool isprime(long n)
{
	long i = 0;
	long h = 0;
	long r = 0;

	if( n<2 )
	{
		return false;
	}
	else if( n==2)
	{
		return true;
	}
	else
	{
		h = n/2+1;
		for(i=2; i<h; i++)
		{
			r = n%i;	
			if(r==0) return false;
		}
		
		return true;
	}
}
